test/scripts/fullNetworkSetup.ts

```diff
@@ -0,0 +1,149 @@
+    console.log("Executor parameters are already set to ideal values ✅");
+  } else {
+    // Increasing times for the relay chain
+    const setConfig = relayApi.tx.Configuration.set_executor_params({
+      new: [
+        { type: "MaxMemoryPages", value: 8192 },
+        {
+          type: "PvfExecTimeout",
+          value: [{ type: "Backing", value: undefined }, 2500n],
+        },
+        {
+          type: "PvfExecTimeout",
+          value: [{ type: "Approval", value: undefined }, 15000n],
+        },
+      ],
+    }).decodedCall;
+
+    // Setting Async Config
+    process.stdout.write("Setting Executor Parameters config for relay chain... ");
+    await relayApi.tx.Sudo.sudo({ call: setConfig }).signAndSubmit(accounts.alice.sr25519.signer);
+    process.stdout.write("✅\n");
+  }
+
+  await waitForChain(shClient);
+
+  // Settings Balances
+  const {
+    data: { free },
+  } = await storageApi.query.System.Account.getValue(accounts["sh-BSP"].sr25519.id);
+
+  if (free < 1_000_000_000_000n) {
+    const setbalance = storageApi.tx.Balances.force_set_balance({
+      who: MultiAddress.Id(accounts["sh-BSP"].sr25519.id),
+      new_free: 1000_000_000_000_000_000n,
+    }).decodedCall;
+
+    const setbalance2 = storageApi.tx.Balances.force_set_balance({
+      who: MultiAddress.Id(accounts["sh-collator"].sr25519.id),
+      new_free: 1000_000_000_000_000_000n,
+    }).decodedCall;
+
+    process.stdout.write("Using sudo to increase BSP account balance... ");
+
+    const { nonce } = await storageApi.query.System.Account.getValue(accounts.alice.sr25519.id);
+    const tx1 = storageApi.tx.Sudo.sudo({ call: setbalance }).signAndSubmit(
+      accounts.alice.sr25519.signer,
+      { nonce }
+    );
+
+    const tx2 = storageApi.tx.Sudo.sudo({ call: setbalance2 }).signAndSubmit(
+      accounts.alice.sr25519.signer,
+      { nonce: nonce + 1 }
+    );
+
+    await Promise.all([tx1, tx2]);
+
+    process.stdout.write("✅\n");
+    const {
+      data: { free },
+    } = await storageApi.query.System.Account.getValue(accounts["sh-BSP"].sr25519.id);
+
+    console.log(`BSP account balance reset by sudo, new free is ${free / 10n ** 12n} balance ✅`);
+  } else {
+    console.log(`BSP account balance is  already ${free / 10n ** 12n} balance ✅`);
+  }
+  // Enrolling BSP
+  const string = "0x8e6a748e6d787260f47f61df1e2cac065db8c1d41428eb178102177876071c6b";
+  const buffer = Buffer.from(string, "utf8");
+  const uint8Array = new Uint8Array(buffer);
+
+  process.stdout.write(`Requesting sign up for ${accounts["sh-BSP"].sr25519.id} ...`);
+  await storageApi.tx.Providers.request_bsp_sign_up({
+    capacity: 5000000,
+    multiaddresses: [new Binary(uint8Array)],
+  }).signAndSubmit(accounts["sh-BSP"].sr25519.signer); // troubleshoot why this is not working
+  process.stdout.write("✅\n");
+
+  console.log("Waiting for randomness (9 blocks)...");
+  await waitForChain(shClient, { blocks: 9, timeoutMs: 120_000 });
```

Comment:
I assume we would be able to query BABE to get the exact amount of blocks required to wait before randomness is valid.

Reply:
done